registry/reference.go

```diff
@@ -160,16 +163,19 @@ func (r Reference) ValidateRepository() error {
 }
 
 // ValidateReference validates the reference.
-func (r Reference) ValidateReference() error {
-	if r.Reference == "" {
-		return nil
-	}
-	if _, err := r.Digest(); err == nil {
+func (r Reference) ValidateReference(isTag bool) error {
```

Comment:
`isTag` sounds reasonable. Could we have 3 separated explicit methods like
```go
func (r Reference) ValidateReference() error
func (r Reference) ValidateReferenceAsTag() error
func (r Reference) ValidateReferenceAsDigest() error
```
?

Reply:
Is this to reduce cyclomatic complexity?  I've made the change, but not sure if it's what you're expecting; let me know what you think.